File: kt-duitang-DDD失败/kt-base/src/main/kotlin/org/cyk/base/utils/JwtUtils.java
package org.cyk.base.utils;

import com.auth0.jwt.JWT;
import com.auth0.jwt.JWTCreator;
import com.auth0.jwt.algorithms.Algorithm;
import com.auth0.jwt.interfaces.DecodedJWT;

import java.util.Calendar;
import java.util.Map;

/**
 * Jwt 工具类
 */
public class JwtUtils {

    //自定义密钥
    private static final String SIGN = "Y*(GY*G^&*%69g*()&";

    /**
     * 生成 Token
     * @param map 自定义的载荷数据
     * @return 返回 Token
     */
    public static String createToken(Map<String, Long> map) {
        //1.设置过期时间(默认 1 天过期)
        Calendar instance = Calendar.getInstance();
        instance.add(Calendar.DATE, 1);

        //2.创建 jwt builder，添加自定义的载荷数据
        JWTCreator.Builder builder = JWT.create();
        for(Map.Entry<String, Long> entry : map.entrySet()) {
            builder.withClaim(entry.getKey(), entry.getValue());
        }

        //3.生成 Token
        String token = builder.withExpiresAt(instance.getTime()) //过期时间
                .sign(Algorithm.HMAC256(SIGN));// sign
        return token;
    }

    /**
     * 验证 Token 合法性
     * @param token
     */
    public static boolean checkToken(String token) {
        try {
            JWT.require(Algorithm.HMAC256(SIGN)).build().verify(token);
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
        return true;
    }


    /**
     * 获取 Token 信息
     * @param token
     * @return
     */
    public static DecodedJWT getTokenInfo(String token) {
        DecodedJWT verify = JWT.require(Algorithm.HMAC256(SIGN)).build().verify(token);
        return verify;
    }

}

File: kt-duitangDDD重构失败/src/main/kotlin/org/cyk/ktduitang/infra/tools/JwtUtils.kt
package org.cyk.ktduitang.infra.tools

import com.auth0.jwt.JWT
import com.auth0.jwt.algorithms.Algorithm
import com.auth0.jwt.interfaces.DecodedJWT
import java.util.*

/**
 * Jwt 工具类
 */
object JwtUtils {

    //自定义密钥
    private const val SIGN = "Y*(GY*G^&*%69g*()&"
    //过期时间(默认 1 天过期)
    private const val EXPIRE_TIME = 1

    /**
     * 生成 Token
     * @param map 自定义的载荷数据
     * @return 返回 Token
     */
    fun createToken(map: Map<String, String>): String {
        //1.设置过期时间
        val expireTime = Calendar.getInstance().apply {
                this.add(Calendar.DATE, EXPIRE_TIME)
            }.time

        //2.创建 jwt builder，添加自定义的载荷数据
        val builder = JWT.create()
        for ((key, value) in map) {
            builder.withClaim(key, value)
        }

        //3.生成 Token
        return builder.withExpiresAt(expireTime) //过期时间
            .sign(Algorithm.HMAC256(SIGN)) // sign
    }

    /**
     * 验证 Token 合法性
     * @param token
     */
    fun checkToken(token: String): Boolean {
        return try {
            JWT.require(Algorithm.HMAC256(SIGN)).build().verify(token)
            true
        } catch (e: Exception) {
            e.printStackTrace()
            false
        }
    }

    /**
     * 获取 Token 信息
     * @param token
     * @return
     */
    fun getTokenInfo(token: String): DecodedJWT {
        return JWT.require(Algorithm.HMAC256(SIGN)).build().verify(token)
    }
}
